src/solver/optimisation/new_constraint_builder.h

```diff
@@ -0,0 +1,400 @@
+#pragma once
+
+#include <antares/logs/logs.h>
+#include "opt_structure_probleme_a_resoudre.h"
+#include "opt_rename_problem.h"
+#include "opt_fonctions.h"
+#include <memory>
+// TODO remove relative include
+#include "../simulation/sim_structure_probleme_economique.h"
+
+#include <variant>
+#include <utility>
+
+namespace NewVariable
+{
+struct SingleIndex
+{
+    explicit SingleIndex(unsigned index) : index(index)
+    {
+    }
+    unsigned index;
+};
+
+struct DispatchableProduction : public SingleIndex
+{
+    using SingleIndex::SingleIndex;
+};
+struct NODU : public SingleIndex
+{
+    using SingleIndex::SingleIndex;
+};
+struct NumberStoppingDispatchableUnits : public SingleIndex
+{
+    using SingleIndex::SingleIndex;
+};
+struct NumberStartingDispatchableUnits : public SingleIndex
+{
+    using SingleIndex::SingleIndex;
+};
+struct NumberBreakingDownDispatchableUnits : public SingleIndex
+{
+    using SingleIndex::SingleIndex;
+};
+
+struct NTCDirect : public SingleIndex
+{
+    using SingleIndex::SingleIndex;
+};
+struct IntercoDirectCost : public SingleIndex
+{
+    using SingleIndex::SingleIndex;
+};
+struct IntercoIndirectCost : public SingleIndex
+{
+    using SingleIndex::SingleIndex;
+};
+
+struct ShortTermStorageInjection : public SingleIndex
+{
+    using SingleIndex::SingleIndex;
+};
+struct ShortTermStorageWithdrawal : public SingleIndex
+{
+    using SingleIndex::SingleIndex;
+};
+struct ShortTermStorageLevel : public SingleIndex
+{
+    using SingleIndex::SingleIndex;
+};
+
+struct HydProd : public SingleIndex
+{
+    using SingleIndex::SingleIndex;
+};
+struct HydProdDown : public SingleIndex
+{
+    using SingleIndex::SingleIndex;
+};
+struct HydProdUp : public SingleIndex
+{
+    using SingleIndex::SingleIndex;
+};
+struct Pumping : public SingleIndex
+{
+    using SingleIndex::SingleIndex;
+};
+struct HydroLevel : public SingleIndex
+{
+    using SingleIndex::SingleIndex;
+};
+struct Overflow : public SingleIndex
+{
+    using SingleIndex::SingleIndex;
+};
+struct FinalStorage : public SingleIndex
+{
+    using SingleIndex::SingleIndex;
+};
+struct PositiveUnsuppliedEnergy : public SingleIndex
+{
+    using SingleIndex::SingleIndex;
+};
+struct NegativeUnsuppliedEnergy : public SingleIndex
+{
+    using SingleIndex::SingleIndex;
+};
+
+struct LayerStorage
+{
+    LayerStorage(unsigned area, unsigned layer) : area(area), layer(layer)
+    {
+    }
+    unsigned area, layer;
```

Comment:
We still need to have a style guide, but I think usually it's preferred to have each member on its own line.

Reply:
Thank you, I note your comment, but this code will soon be obsolete. #1607